pytest-embedded/pytest_embedded/dut_factory.py

```diff
@@ -408,6 +412,21 @@ def serial_gn(_fixture_classes_and_options, msg_queue, app) -> t.Optional[t.Unio
     kwargs = _fixture_classes_and_options.kwargs['serial']
     if 'app' in kwargs and kwargs['app'] is None:
         kwargs['app'] = app
+
+    if kwargs.get('flash_port'):
+        operation_port = kwargs.pop('port', None)
+        if operation_port is None:
+            raise SystemExit('If the flash port was set up, the port should also be set up.')
+
+        flash_port = kwargs.pop('flash_port')
+        kwargs['stop_after_init'] = True
+        kwargs['port'] = flash_port
+        cls(**_drop_none_kwargs(kwargs))
```

Comment:
this instance has not been returned, so the teardown won't happen in `multi_dut_generator_fixture`.

in `multi_dut_generator_fixture`, we only teardown the obj that returns.

```python
...
            try:
                res = func(*args, **kwargs)
                yield res
            finally:
                if res:
                    _close_or_terminate(res)
```

Reply:
@hfudev Hello, you are right about multiple port occupation for dut's. Therefore, I have updated the occupation logic. Please take a look.